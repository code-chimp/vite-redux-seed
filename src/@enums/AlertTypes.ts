enum AlertTypes {
  Error = 'danger',
  Info = 'info',
  Success = 'success',
  Warning = 'warning',
}

export default AlertTypes;
