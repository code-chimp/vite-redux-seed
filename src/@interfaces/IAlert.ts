import AlertType from '../@types/AlertType';

interface IAlert {
  id: string;
  type: AlertType;
  text?: string;
  title?: string;
}

export default IAlert;
