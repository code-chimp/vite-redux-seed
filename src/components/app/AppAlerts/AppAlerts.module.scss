.wrapper {
  width: 100%;
  display: flex;
  position: fixed;
  justify-content: center;
  top: 30px;
  z-index: 9999;
}

.alertsCol {
  width: 50%;
  position: absolute;
  white-space: pre-line;
}
