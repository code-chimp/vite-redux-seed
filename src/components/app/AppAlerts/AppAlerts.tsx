import React, { FC } from 'react';
import IAlert from '../../../@interfaces/IAlert';
import { useAppSelector } from '../../../helpers';
import { selectAlerts } from '../../../store/slices/alerts';
import styles from './AppAlerts.module.scss';
import Alert from './Alert';

const AppAlerts: FC = () => {
  const alerts = useAppSelector(selectAlerts);

  return (
    <div className={styles.wrapper}>
      <div className={styles.alertsCol}>
        {alerts.map((alert: IAlert) => (
          <Alert key={alert.id} alert={alert} />
        ))}
      </div>
    </div>
  );
};

export default AppAlerts;
