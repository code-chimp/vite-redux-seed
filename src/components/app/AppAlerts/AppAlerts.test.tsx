import React from 'react';
import { render } from '@testing-library/react';
import configureMockStore, { MockStore } from 'redux-mock-store';
import { Provider } from 'react-redux';
import IAlert from '../../../@interfaces/IAlert';
import AlertTypes from '../../../@enums/AlertTypes';
import AppAlerts from './AppAlerts';

const mockStore = configureMockStore([]);
let store: MockStore;

const initialState: Array<IAlert> = [
  {
    id: 'foo',
    text: 'batten down the hatches',
    type: AlertTypes.Warning,
  },
];

describe('components / app / AppAlerts', () => {
  it('should match the snapshot', () => {
    store = mockStore({ alerts: initialState });
    const { asFragment } = render(
      <Provider store={store}>
        <AppAlerts />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
