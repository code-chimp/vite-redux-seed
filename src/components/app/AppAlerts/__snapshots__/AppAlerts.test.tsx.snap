// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components / app / AppAlerts > should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="_wrapper_2fe02d"
  >
    <div
      class="_alertsCol_2fe02d"
    >
      <div
        class="alert alert-warning alert-dismissible fade show d-flex align-items-center"
        data-testid="alert-foo"
        role="alert"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-triangle-exclamation flex-shrink-0 me-2"
          data-icon="triangle-exclamation"
          data-prefix="fas"
          data-testid="alert-foo-icon"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7 .2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8 .2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24V296c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0 -64 0 32 32 0 1 0 64 0z"
            fill="currentColor"
          />
        </svg>
        <div>
          <span
            data-testid="alert-foo-text"
          >
            batten down the hatches
          </span>
        </div>
        <button
          aria-label="Close"
          class="btn-close"
          data-bs-dismiss="alert"
          data-testid="alert-foo-close"
          type="button"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
